explorer/interpreter/type_checker.cpp

```diff
@@ -1939,43 +1944,54 @@ auto TypeChecker::TypeCheckExp(Nonnull<Expression*> e,
         ts.push_back(&argument->static_type());
       }
       switch (op.op()) {
-        case Operator::Neg:
-          CARBON_RETURN_IF_ERROR(ExpectExactType(e->source_loc(), "negation",
-                                                 arena_->New<IntType>(), ts[0],
-                                                 impl_scope));
-          op.set_static_type(arena_->New<IntType>());
-          op.set_value_category(ValueCategory::Let);
+        case Operator::Neg: {
+          // Handle a built-in negation first.
```

Comment:
Just to note, I'm fine with this here, but per my comment on the Eq PR I do think we should be switching to intrinsics for basic types instead of this approach. I think the uniformity will be helpful, particularly as we work towards generalizing support for i8, u32, f64, etc.